docs/usage/routing/handlers.rst

```diff
@@ -2,320 +2,331 @@ Route handlers
-* ``query`` : injects the request ``query_params`` as a parsed dictionary.
-* ``request``: injects the :class:`Request <.connection.Request>` instance. Available only for `http route handlers`_
-* ``scope`` : injects the ASGI scope dictionary.
+* ``cookies``: injects the request :class:`cookies <.datastructures.cookie.Cookie>` as a parsed
+  :class:`dictionary <dict>`.
+* ``headers``: injects the request headers as a parsed :class:`dictionary <dict>`.
+* ``query`` : injects the request ``query_params`` as a parsed :class:`dictionary <dict>`.
+* ``request``: injects the :class:`Request <.connection.Request>` instance. Available only for `HTTP route handlers`_
+* ``scope`` : injects the ASGI scope :class:`dictionary <dict>`.
 * ``socket``: injects the :class:`WebSocket <.connection.WebSocket>` instance. Available only for `websocket route handlers`_
 * ``state`` : injects a copy of the application :class:`State <.datastructures.state.State>`.
-* ``body`` : the raw request body. Available only for `http route handlers`_
+* ``body`` : the raw request body. Available only for `HTTP route handlers`_
 
-Note that if your parameters collide with any of the reserved keyword arguments above, you can :ref:`provide an alternative name <usage/routing/parameters:Alternative names and constraints>`.
+Note that if your parameters collide with any of the reserved :term:`keyword arguments <argument>` above, you can
+:ref:`provide an alternative name <usage/routing/parameters:Alternative names and constraints>`.
 
 For example:
 
 .. code-block:: python
+    :caption: Providing an alternative name for a reserved keyword argument
 
-   from typing import Any, Dict
-   from litestar import Request, get
-   from litestar.datastructures import Headers, State
+    from typing import Any, Dict
+    from litestar import Request, get
+    from litestar.datastructures import Headers, State
 
 
-   @get(path="/")
-   async def my_request_handler(
+    @get(path="/")
+    async def my_request_handler(
        state: State,
        request: Request,
        headers: Dict[str, str],
        query: Dict[str, Any],
        cookies: Dict[str, Any],
-   ) -> None: ...
+    ) -> None: ...
 
-.. tip::
-
-    You can define a custom typing for your application state and then use it as a type instead of just using the
-    State class from Litestar
+.. tip:: You can define a custom typing for your application state and then use it as a type instead of just using the
+    :class:`~.datastructures.state.State` class from Litestar
 
 Type annotations
 ----------------
 
-Litestar enforces strict type annotations. Functions decorated by a route handler **must** have all their kwargs and
-return value type annotated. If a type annotation is missing, an
-:class:`ImproperlyConfiguredException <litestar.exceptions.ImproperlyConfiguredException>` will be raised during the
+Litestar enforces strict type annotations. Functions decorated by a route handler **must** have all their
```

Comment:
I wonder if there's a link we can use for the term "annotations" like we've done for other things like decorators etc. Maybe https://peps.python.org/pep-0484/#type-definition-syntax?

Reply:
Im not sure here, i believe i tried but the base PR this comes from is a little old so i may be misremembering.

Reply:
oh not i didnt try at all or hard enough (or both)
Is this what you mean?
https://docs.python.org/3/glossary.html#term-annotation

Reply:
Yeh that'll do!